src/Interfaces/AbstractHyperrectangle.jl

```diff
@@ -189,7 +189,7 @@ function vertices_list(H::AbstractHyperrectangle{N}) where {N<:Real}
     trivector = Vector{Int8}(undef, n)
     m = 1
     c = center(H)
-    v = copy(c)
+    v = convert(Vector{N}, copy(c))
```

Comment:
```suggestion
    VN = typeof(radius_hyperrectangle(H, 1))
    v = convert(VN, copy(c))
```

As an example why this is better, we may change the code below to return a sparse vector in the future.
https://github.com/JuliaReach/LazySets.jl/blob/894f77bd2309a80e02d5be50acfb6dbc7c617947/src/Interfaces/AbstractSingleton.jl#L67-L69

Reply:
Hmm, I disagree. This is a generic fallback and `AbstractHyperrectangle{N}` does not have the radius/center as type parameters that we can readily use. Moreover the algorithm for sparse would be another one, see lines 192-200 and 209-221, we should iterate and fill the vectors differently, etc.

Reply:
in other words, what i meant to say is that the implementation for sparse vectors requires other more fundamental changes in this function.